packages/components/src/components/atoms/form/RequestTypeItem.tsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react'
+import { styled } from '@mui/system'
+
+import {
+  ILimitations,
+  ILimitationsTypes,
+  IOptionsTags,
+  IRequestType,
+  IRequestTypesOptions,
+} from '@elastic-suite/gally-admin-shared'
+
+import TextFieldTagsMultiple from './TextFieldTagsMultiple'
+
+import Checkbox from './Checkbox'
+import { IconButton } from '@mui/material'
+import IonIcon from '../IonIcon/IonIcon'
+import {
+  CustomFirstSelectedItem,
+  CustomRootItem,
+  CustomSelectedItem,
+} from './RequestTypeItem.styled'
+
+const CustomItem = styled('div')(({ theme }) => ({
+  display: 'flex',
+  gap: theme.spacing(3),
+  alignItems: 'start',
+}))
+
+const CustomLabel = styled('div')(({ theme }) => ({
+  color: theme.palette.colors.neutral[900],
+  fontFamily: 'var(--gally-font)',
+  fontSize: theme.spacing(1.5),
+  fontWeight: '600',
+  width: '190px',
+  lineHeight: '18px',
+}))
+
+const CustomDataLimitations = styled('div')(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'row',
+  gap: theme.spacing(2),
+  alignItems: 'center',
+}))
+
+export interface ITextFieldTagssss {
+  value: IRequestType
+  onChange: (value: IRequestType) => void
+  options: IOptionsTags[]
+  limitationsTypes: ILimitationsTypes[]
+  requestTypesOptions: IRequestTypesOptions[]
+}
+
+function RequestTypeItem(props: ITextFieldTagssss): JSX.Element {
+  const { value, onChange, options, limitationsTypes, requestTypesOptions } =
+    props
+
+  function onChangeVal(
+    idItem: IRequestTypesOptions[] | ILimitations[],
+    val?: boolean | string
+  ): void {
+    if (Array.isArray(idItem) && typeof val === 'boolean') {
+      const newData = value.requestTypes.map((item) => {
+        if (
+          item.requestType ===
+          (idItem as IRequestTypesOptions[]).find(
+            (it) => it.value === item.requestType
+          )?.value
+        ) {
+          return { ...item, applyToAll: !item.applyToAll }
+        }
+        return item
+      })
+      return onChange({ ...value, requestTypes: newData })
+    }
+
+    if (Array.isArray(idItem) && typeof val === 'string') {
+      return onChange({ ...value, [val]: idItem })
+    }
+
+    if (!val) {
+      const newData = value.requestTypes.filter(
+        (item) =>
+          item.requestType !==
+          (idItem as IRequestTypesOptions[]).find(
+            (it) => it.value === item.requestType
+          )?.value
+      )
+
+      return onChange({ ...value, requestTypes: newData })
+    }
+  }
+
+  function countLines(): number {
+    const listUniqueLimitationType: string[] = []
+    for (const item of value.requestTypes) {
+      const limitation_type = requestTypesOptions.find(
+        (its) => its.value === item.requestType
+      )?.limitation_type
+
+      if (
+        limitation_type &&
+        !listUniqueLimitationType.includes(limitation_type)
+      ) {
+        listUniqueLimitationType.push(limitation_type)
+      }
+    }
+    return listUniqueLimitationType.length
+  }
```

Comment:
Pas besoin de faire une fonction que tu appelles ensuite 3 fois.
Pourquoi exécuter 3 fois le même code pour obtenir 3 fois le même résultats ?
De plus faire un `find` dans un `for` c'est faire un boucle dans une boucle et donc mauvais d'un point de vue algorithme.
Quelque chose comme ça devrait fonctionner:
```ts
const limitationTypeMap = requestTypesOptions.reduce<Record<string, string>>((acc, option) => {
  acc[option.value] = option.limitation_type
  return acc
}, {});
const uniqLimitationTypes = new Set(value.requestTypes.map(item => limitationTypeMap[item.requestType]))
const countLines =  uniqLimitationTypes.sizes
```